### ISSUE TITLE
Persistant Issue with UnityEnvironment() for Custom Training Algorithms

### ISSUE THREAD

**Issue by justkittenaround** [Unity-Technologies/ml-agents#4545]
**Describe the bug**
Cannot load custom environment with ml-agents-envs API via UnityEnvironment(). _**I would like to load my custom environment I built in Unity editor and access it with the ml-agents API so I can train it with my custom learning algorithm in python.**_ 

**To Reproduce**
1. Go to https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Installation.md and follow download instructions (not advanced, just regular).
2. Go to https://github.com/Unity-Technologies/ml-agents/blob/release_7_docs/docs/Learning-Environment-Executable.md and follow the instructions until "Training the environment'. **Note** 'Display Resolution Dialog' does not exist but is recommended on the docs.
4. See error in terminal.
* or you can watch the video I made of the error here: https://youtu.be/btBZCJ3w25M

**Console logs / stack traces**
`>>>import mlagents`
`>>>from mlagents_envs.environment import UnityEnvironment as UE`

`>>>env = UE(file_name = '3DBall_build')`
`Found path: /home/whale/Unity/ml-agents-release_7/3DBall_build.x86_64`

`>>>env`
`<mlagents_envs.environment.UnityEnvironment object at 0x7fbc4a103eb8>`

 `>>>list(env.behavior_specs)`
`[]`

**Screenshots**
see video under reproduce section above

**Environment (please complete the following information):**
 I have tried with multiple Unity environments, both my own that I built and examples from the latest release (7) in ml-agents repository. I've also tried uninstalling everything and reinstalling. I've tried with Anaconda and just on local. 
- Unity Version: 2019.4.11f1
- OS + version: Ubuntu 18.04 LTS
- _ML-Agents version_: ML-Agents v0.20.0
- _ML-agents package version in the Unity editor_: 1.4.0.
- _TensorFlow version_: 1.15.0
- _Environment_: 3DBall, Crawler, RollerBall, custom

**Comment by andrewcoh** [comment]
Hi @justkittenaround 

Can you try calling `env.reset()` and then checking this list? Please see my screenshot:

<img width="929" alt="Screen Shot 2020-10-09 at 10 13 18 AM" src="https://user-images.githubusercontent.com/54679309/95593938-4949d380-0a18-11eb-92a1-cdd3cb0594a1.png">

**Comment by justkittenaround** [comment]
Hi @andrewcoh, thanks for looking into it. env.reset() did not resolve the error. Bellow is the results. 
![Screenshot from 2020-10-09 10-39-41](https://user-images.githubusercontent.com/36868866/95596996-16094380-0a1c-11eb-905f-9489040df50b.png)

**Comment by andrewcoh** [comment]
Yep, I'm aware you're using Ubuntu.

I should've asked this earlier (but I think I saw your forum post mentioning this): are you able to get this to run using `mlagents-learn` and the --env command?

**Comment by justkittenaround** [comment]
@andrewcoh  thanks for taking that into account. I did originally post on the forum because I thought it might be something I was doing wrong. 

I am able to get it training using the mlagents-learn command. The problem is that I want to use my own training algorithm instead of the built in ones. I was under the impression that we can't do that with mlagents-learn, we need to use the mlagents-envs. Is that correct?

**Comment by andrewcoh** [comment]
Yes, you have that correct and it's not obvious to me just yet that you're doing anything wrong. I'll do some troubleshooting and get back to you.

**Comment by justkittenaround** [comment]
@andrewcoh  I seem to have found a solution. First, the build needs to be in the project folder where I created the scene in unity editor. Secondly, I have to build the executable in debugging mode only. Third, when the environment opens after I call UnityEnvironment with the file name, I have to click "wait" when it crashes instead of "force close". 

Thanks for looking into it! Seems a little buggy but glad I got a solution so I can start using the awesome API.
